github_connector/models/abstract_github_model.py

```diff
@@ -0,0 +1,207 @@
+            'github_create_date': data.get('created_at', False),
+            'github_write_date': data.get('updated_at', False),
+            'github_last_sync_date':
+            datetime.now().strftime('%Y-%m-%d %H:%M:%S'),
+        }
+
+    @api.multi
+    def full_update(self):
+        pass
+
+    # Custom Public Function
+    @api.model
+    def get_from_id_or_create(self, data, extra_data=None):
+        """Search if the odoo object exists in database. If yes, returns the
+            object. Otherwise, creates the new object.
+
+            :param data: dict with github 'id' and 'url' keys
+            :return: The searched or created object
+
+            :Example:
+
+            >>> self.env['github_organization'].get_from_id_or_create(
+                {'id': 7600578, 'url': 'https://api.github.com/orgs/OCA'})
+        """
+        extra_data = extra_data and extra_data or {}
+        res = self.search([('github_id', '=', data['id'])])
+        if not res:
+            if self._need_individual_call:
+                github_model = self.get_github_for(self.github_type())
+                data = github_model.get_by_url(data['url'])
+            return self._create_from_github_data(data, extra_data)
+        else:
+            return res
+
+    @api.model
+    def create_from_name(self, name):
+        """Call Github API, using a url using github name. Load data and
+            Create Odoo object accordingly, if the odoo object doesn't exist.
+
+            :param name: the github name to load
+            :return: The created object
+
+            :Example:
+
+            >>> self.env['github_organization'].create_from_name('OCA')
+            >>> self.env['github_repository'].create_from_name('OCA/web')
+        """
+        github_model = self.get_github_for(self.github_type())
+        res = github_model.get([name])
+        # search if ID doesn't exist in database
+        current_object = self.search([('github_id', '=', res['id'])])
+        if not current_object:
+            # Create the object
+            return self._create_from_github_data(res)
+        else:
+            # Manage the special case when the name changed...
+            # TODO
+            pass
+
+    @api.multi
+    def button_update_from_github_light(self):
+        return self.update_from_github(False)
+
+    @api.multi
+    def button_update_from_github_full(self):
+        return self.update_from_github(True)
+
+    @api.multi
+    def update_from_github(self, child_update):
+        """Call Github API, using a url using github id. Load data and
+            update Odoo object accordingly, if the odoo object is obsolete.
+            (Based on last write dates)
+
+            :param child_update: set to True if you want to reload childs
+                Objects linked to this object. (like members for teams)
+        """
+        github_model = self.get_github_for(self.github_type())
+        for item in self:
```

Comment:
It would be nice if this loop diverted to another method in order to allow inherited injection to the conditionals & res